hw_35_generics.ts

```diff
@@ -0,0 +1,66 @@
+// С ниже приведенным массивом решить следующие задачи. Все функции и данные должны быть протипизированы:
+const users2 = [
+    {
+        name: "Harry Felton",
+        phone: "(09) 897 33 33",
+        email: "felton@gmail.com",
+        animals: ["cat"],
+        cars: ["bmw"],
+        hasChildren: false,
+        hasEducation: true            
+    },
+    {
+        name: "May Sender",
+        phone: "(09) 117 33 33",
+        email: "sender22@gmail.com",
+        hasChildren: true,
+        hasEducation: true
+    },
+    {
+        name: "Henry Ford",
+        phone: "(09) 999 93 23",
+        email: "ford0@gmail.com",
+        cars: ["bmw", "audi"],
+        hasChildren: true,
+        hasEducation: false
+    }
+]
+
+// 1. Создать строку из имен пользователей через запятую
+type Name = {
+    name: string
+}
+const getName2 = <T extends Name>(users: T[]): string => users.map(({name}): string => name).join(', ')
```

Comment:
Sometimes it's not necessary to create a whole type/interface just for a single property. 
For example, this would be a good choice too:

`<T extends { name: string }>`